agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/AiComponentInstaller.java

```diff
@@ -280,11 +280,22 @@ private static ApplicationInsightsXmlConfiguration buildXmlConfiguration(Configu
         // configure custom jmx metrics
         ArrayList<JmxXmlElement> jmxXmls = new ArrayList<>();
         for (JmxMetric jmxMetric : config.jmxMetrics) {
-            JmxXmlElement jmxXml = new JmxXmlElement();
-            jmxXml.setName(jmxMetric.name);
-            jmxXml.setObjectName(jmxMetric.objectName);
-            jmxXml.setAttribute(jmxMetric.attribute);
-            jmxXmls.add(jmxXml);
+            if (!jmxMetric.attribute.isEmpty()) {
+                JmxXmlElement jmxXml = new JmxXmlElement();
+                jmxXml.setName(jmxMetric.name);
+                jmxXml.setObjectName(jmxMetric.objectName);
+                jmxXml.setAttribute(jmxMetric.attribute);
+                jmxXmls.add(jmxXml);
+            }
+            else if (!jmxMetric.attributes.isEmpty()) {
+                for (String attribute : jmxMetric.attributes) {
+                    JmxXmlElement jmxXml = new JmxXmlElement();
+                    jmxXml.setName(getJmxDisplayName(jmxMetric.objectName) + attribute);
```

Comment:
you can calculate jmxDisplayName once outside of the for-each loop

Reply:
Right... the name would be the same for all attributes...